android/app/src/main/java/com/sample/manifest/Manifest.kt

```diff
@@ -2,14 +2,20 @@ package com.sample.manifest
 
 import com.squareup.moshi.JsonClass
 
-typealias Components = Map<String, Component>
+typealias Components = Array<Component>
 
 @JsonClass(generateAdapter = true)
 data class Manifest(
     val name: String,
     val displayName: String,
+
+    @Suppress("ArrayInDataClass")
     val components: Components
 )
 
 @JsonClass(generateAdapter = true)
-data class Component(val displayName: String?)
+data class Component(
+    val appKey: String,
+    val displayName: String?,
+    val initialProperties: Map<String, String>?
```

Comment:
Can the prop value be `null`? If yes, the type of `initialProperties` needs to reflect that: `Map<String, String?>?`